ParkingLot/src/main/java/services/ExitGate.java

```diff
@@ -1,28 +0,0 @@
-package services;
-
-import dao.TicketsDao;
-import factories.CostComputationFactory;
-import factories.ParkingSpotManagerFactory;
-import models.Ticket;
-
-public class ExitGate {
-    private final ParkingSpotManagerFactory parkingSpotManagerFactory;
-    private final CostComputationFactory costComputationFactory;
-    private final TicketsDao ticketsDao;
-
-    public ExitGate(ParkingSpotManagerFactory parkingSpotManagerFactory, CostComputationFactory costComputationFactory, TicketsDao ticketsDao) {
-        this.parkingSpotManagerFactory = parkingSpotManagerFactory;
-        this.costComputationFactory = costComputationFactory;
-        this.ticketsDao = ticketsDao;
-    }
-
-    public long calculateCost(Ticket ticket) {
-        return costComputationFactory.getCostComputationByVehicleType(ticket.getVehicle().getVehicleType()).calculateCost(ticket);
-    }
-
-    public void processExit(Ticket ticket) {
-        parkingSpotManagerFactory.getParkingSpotManagerByVehicleType(ticket.getVehicle().getVehicleType()).freeSpot(ticket.getAssignedParkingSpot());
```

Comment:
shouldn't this be in the dao layer?

Reply:
We have only created DAO for the tickets here. The parking spots are being managed by the SpotManager classes. If we add doa layer for the parking spots, then the final data update should happen in dao layer.